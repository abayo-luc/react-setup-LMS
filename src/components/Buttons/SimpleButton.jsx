import React from 'react';
import PropTypes from 'prop-types';

const SimpleButton = (props) => {
  const { title, onClick, disabled } = props;
  return (
    <button
      type="button"
      className="btn btn-primary btn-lg btn-block"
      onClick={onClick}
      disabled={disabled}
    >
      {title}
    </button>
  );
};
SimpleButton.propTypes = {
  title: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
};
SimpleButton.defaultProps = {
  onClick: () => null,
  disabled: false,
};
export default SimpleButton;
