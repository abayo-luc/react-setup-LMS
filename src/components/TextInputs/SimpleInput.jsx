import React from 'react';
import PropTypes from 'prop-types';
import './input.scss';

const SimpleInput = (props) => {
  const {
    type, name, value, id, onChange, placeholder, label,
  } = props;
  return (
    <label htmlFor="inputEmail4">
      {label}

      <input
        type={type}
        className="form-control"
        name={name}
        id={id}
        placeholder={placeholder}
        value={value}
        onChange={onChange}
      />
    </label>
  );
};
SimpleInput.propTypes = {
  type: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  id: PropTypes.string,
  value: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};
SimpleInput.defaultProps = {
  id: '',
};
export default SimpleInput;
