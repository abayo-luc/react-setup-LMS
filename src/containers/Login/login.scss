#login {
  height: 100vh;
  background: linear-gradient(
      180deg,
      rgba(0, 123, 255, 0.8) 0%,
      rgba(23, 23, 22, 0) 97.24%
    ),
    url('../../assets/img/rwanda.jpg');
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;

  display: grid;
  form {
    width: 25%;
    margin: auto;
    display: flex;
    flex-direction: column;
  }
  @media (max-width: 768px) {
    form {
      width: 80%;
    }
  }
}
