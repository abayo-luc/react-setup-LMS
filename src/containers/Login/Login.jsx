import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import SimpleInput from '../../components/TextInputs/SimpleInput';
import './login.scss';
import SimpleButton from '../../components/Buttons/SimpleButton';
import { handleOnChange, handleLogin } from '../../store/actions/auth';

export class Login extends Component {
  state = {};

  handleSubmit = () => {
    const { handleLogin: sendCredential, email, password } = this.props;
    sendCredential({ email, password });
  };

  render() {
    const {
      handleOnChange: handleOnChangeText,
      email,
      password,
      isSubmitting,
    } = this.props;
    return (
      <div id="login">
        <form>
          <SimpleInput
            type="email"
            placeholder="Email"
            name="email"
            label="Email"
            value={email}
            onChange={e => handleOnChangeText(e.target.name, e.target.value)}
          />
          <SimpleInput
            type="password"
            placeholder="Password"
            name="password"
            label="Password"
            value={password}
            onChange={e => handleOnChangeText(e.target.name, e.target.value)}
          />
          <SimpleButton
            title="Login"
            onClick={this.handleSubmit}
            disabled={isSubmitting}
          />
        </form>
      </div>
    );
  }
}

Login.propTypes = {
  handleOnChange: PropTypes.func.isRequired,
  email: PropTypes.string,
  password: PropTypes.string,
  isSubmitting: PropTypes.bool,
  handleLogin: PropTypes.func.isRequired,
  isSuccess: PropTypes.func.bool,
  message: PropTypes.string,
};
Login.defaultProps = {
  email: '',
  password: '',
  isSubmitting: false,
  isSuccess: false,
  message: '',
};
const mapStateToProps = (state) => {
  const { auth } = state;
  return {
    ...auth,
  };
};
export default connect(
  mapStateToProps,
  { handleOnChange, handleLogin },
)(Login);
