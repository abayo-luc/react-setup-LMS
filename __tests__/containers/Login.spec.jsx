import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Login } from '../../src/containers/Login/Login';
import SimpleButton from '../../src/components/Buttons/SimpleButton';
import SimpleInput from '../../src/components/TextInputs/SimpleInput';

const [handleOnChange, handleLogin] = new Array(2).fill(jest.fn());
const props = {
  handleOnChange,
  handleLogin,
  email: '',
  password: '',
  isSubmitting: false,
};
const warper = shallow(<Login {...props} />);
describe('Login component', () => {
  beforeEach(() => {});
  describe('component rendered', () => {
    test('should match the snapshot', () => {
      expect(toJson(warper)).toMatchSnapshot();
    });
  });
  describe('component methods', () => {
    let instance;
    beforeEach(() => {
      instance = warper.instance();
      jest.spyOn(instance, 'handleSubmit');
    });
    afterEach(() => {
      instance.handleSubmit.mockClear();
      handleOnChange.mockClear();
      handleLogin.mockClear();
    });
    test('should handle email field input change', () => {
      warper
        .find(SimpleInput)
        .at(0)
        .simulate('change', {
          target: {
            name: 'email',
            value: 'jean.abayo@gmail.com',
          },
        });
      expect(handleOnChange).toBeCalledWith('email', 'jean.abayo@gmail.com');
    });
    test('should handle password field input change', () => {
      warper
        .find(SimpleInput)
        .at(1)
        .simulate('change', {
          target: {
            name: 'password',
            value: '12345',
          },
        });
      expect(handleOnChange).toBeCalledWith('password', '12345');
    });
    test('should dispatch login action', () => {
      const payload = {
        email: 'jean.abayo@gmail.com',
        password: 'password',
      };
      warper.setProps({ ...payload });
      warper
        .find(SimpleButton)
        .at(0)
        .simulate('click');
      expect(handleLogin).toBeCalledWith({
        ...payload,
      });
    });
  });
});
