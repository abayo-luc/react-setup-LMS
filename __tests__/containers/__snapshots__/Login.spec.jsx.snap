// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component component rendered should match the snapshot 1`] = `
<div
  id="login"
>
  <form>
    <SimpleInput
      id=""
      label="Email"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="email"
      value=""
    />
    <SimpleInput
      id=""
      label="Password"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <SimpleButton
      disabled={false}
      onClick={[Function]}
      title="Login"
    />
  </form>
</div>
`;
